commcare_connect/opportunity/views.py

```diff
@@ -481,6 +483,34 @@ def payment_delete(request, org_slug=None, opp_id=None, access_id=None, pk=None)
     return redirect("opportunity:user_payments_table", org_slug=org_slug, opp_id=opp_id, pk=access_id)
 
 
+@org_member_required
+def user_profile(request, org_slug=None, opp_id=None, pk=None):
+    access = get_object_or_404(OpportunityAccess, pk=pk, accepted=True)
+    user_visits = UserVisit.objects.filter(user=access.user, opportunity=access.opportunity)
+    user_visit_data = []
+    for user_visit in user_visits:
+        if not user_visit.location:
+            continue
+        lat, lng, elevation, precision = user_visit.location.split(" ")
+        user_visit_data.append(
+            dict(entity_name=user_visit.entity_name, visit_date=user_visit.visit_date.date(), lat=lat, lng=lng)
+        )
+    # user for centering the User visits map
+    lat_avg = reduce(lambda x, y: x + float(y["lat"]), user_visit_data, 0.0) / len(user_visit_data)
```

Comment:
This will cause issues if `user_visit_data` is empty (no visists or none with locations)

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/248/commits/17fbe13cd0b2b74c3388eac42cbf5283ad04273f